circle.yml

```diff
@@ -44,11 +46,17 @@ deployment:
   docker:
     branch: master
     commands:
-      # Build small deploy container for master branch.
       - sed "s/<EMAIL>/$DOCKER_EMAIL/;s/<AUTH>/$DOCKER_AUTH/" < "resources/deploy_templates/.dockercfg.template" > ~/.dockercfg
       - |
+          VERSION=$(git describe 2>/dev/null || git rev-parse --short HEAD 2>/dev/null)
```

Comment:
Yes, that stackoverflow question is what I was referring to. Started to work on a change today but didn't get it finished. `go generate` could certainly be used but seems slightly more awkward to fit into our `go build` that using the ldflags.